File: learning/treat_data.py
#!/usr/bin/python

import sys

def treat_data(fileLoc, featLoc):
    """read data from csv and simplify them"""
    extBase = []
    with open(featLoc) as featFile:
        for line in featFile.readlines():
            extBase.append(line.split(":")[0])
   
    #print "extensions loaded"
    with open(fileLoc) as fileFile:
        for line in fileFile.readlines():
            line = line.split("\n")[0]
            line = line[1:]
            with open('constant.csv', 'a+') as constFile:
                #print "csv file loaded"
                constFile.write(line)
                for ext in extBase:
                    with open("data/" + line + "." + ext + ".csv") as f:
                        #print ext + " csv file loaded"
                        c = 0
                        r = []
            	        for row in f.readlines():
            	            if row[0] != '%':
                                row = row.split(',')
                                c = c + 1
                                try:
                                    for i in xrange(len(row)):
                                        r[i] = r[i] + float(row[i])
                                except:
                                    r = [float(i) for i in row]
            	        r = [i / c for i in r]
            	        for i in r:
            	            constFile.write('$'+str(i))
        	        
                constFile.write('\n')
            

#print treat_data("", "featureplan")
treat_data(sys.argv[1], sys.argv[2])

File: src/learning/treat_data.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
using namespace std;

unsigned int split(const std::string &txt, std::vector<std::string> &strs, char ch)
{
    unsigned int pos = txt.find( ch );
    unsigned int initialPos = 0;
    strs.clear();

    // Decompose statement
    while( pos != std::string::npos ) {
        strs.push_back( txt.substr( initialPos, pos - initialPos + 1 ) );
        initialPos = pos + 1;

        pos = txt.find( ch, initialPos );
    }

    // Add the last one
    strs.push_back( txt.substr( initialPos, std::min( pos, txt.size() ) - initialPos + 1 ) );

    return strs.size();
}

int main(int argc, char** argv) {
    char* fileLoc(argv[0]);
    char* featLoc(argv[1]);

    ifstream reader(featLoc);

    string line;
    vector<string> extBase;     
    vector<string> tmp;

    while(getline(reader, line)) { 
        split(tmp, line, ":");
        extBase.push_back(tmp[0]);
    }
   
    cout << "extensions loaded" << endl;
 
    ofstream constFile('constant.csv');
    constFile << fileLoc;
    
    for(int i = extBase.begin(); i < extBase.end(); i++) {
        ifstream fileTmp((fileLoc + "." + extBase[i] + ".csv").c_str());
        cout << extBase[i] << " csv file loaded" << endl;
        double c(0);
        string row;
        vector<double> r;
        while(getline(fileTmp, row)) {            
            if(row[0] != '%') {
                split(tmp, row, ',');
                c++;
                try
                    for(int j = 0; j < tmp.size(); j++)
                        r[j] += double(row[j]) 
                catch()
                    for(int j = 0; j < tmp.size(); j++)
                        r.push_back(double(row[j]));
            }
        }
        r = (1 / c)*r;
    
        for(int j = r.begin(); j < r.end(); j++)
            constFile << "$" << r[j];
    }
    constFile << endl;
    
    return 0;
}
